app/actions.js

```diff
@@ -444,19 +444,21 @@ const initializeCabal = ({ addr, username, settings }) => async dispatch => {
     }, {
       name: 'init',
       action: () => {
-        const users = cabalDetails.getUsers()
-        const username = cabalDetails.getLocalName()
-        const channels = cabalDetails.getChannels()
-        const channelsJoined = cabalDetails.getJoinedChannels() || []
-        const channelMessagesUnread = getCabalUnreadMessagesCount(cabalDetails)
-        const currentChannel = cabalDetails.getCurrentChannel()
-        const channelMembers = cabalDetails.getChannelMembers()
-        dispatch({ type: 'UPDATE_CABAL', initialized: true, addr, channelMessagesUnread, users, username, channels, channelsJoined, currentChannel, channelMembers })
-        dispatch(getMessages({ addr, amount: 1000, channel: currentChannel }))
-        dispatch(updateAllsChannelsUnreadCount({ addr, channelMessagesUnread }))
-
-        dispatch(viewCabal({ addr, channel: settings.currentChannel }))
-        client.focusCabal(addr)      
+        setTimeout(() => {
```

Comment:
@cblgh also just fyi, i found that when i was building `cabal-ui` that the init event would fire but calling `cabalDetails.getUsers()` immediately inside it would return an empty set of users/channels/etc. if i waited a few seconds, then `cabalDetails.getUsers()` would have content.... so that's what this delay is here for

Reply:
huh! that is definitely strange, do you mind cross-posting this as an issue to cabal-client?

Reply:
ideally this should be a 'ready' event from some lower-level module, i bet it will need to come from the indexer -- which should already be there somewhere. a timeout will give different results on different computers, slow computers or those with spinning disks might not be able to complete the necessary steps in 2 seconds with large cabals